src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -312,6 +312,13 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     if (materializedDocument.TryGetValue("errmsg", out errmsgBsonValue) && errmsgBsonValue.IsString)
                     {
                         var errmsg = errmsgBsonValue.ToString();
+                        if (materializedDocument.TryGetValue("code", out var errorCode)
+                            && errorCode.AsInt32 == 20
```

Comment:
Maybe ToInt32() is safer in case server sometimes returns a different type of number.

Reply:
Ah true, fixed!